dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/python/PythonUdfManagerTest.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    @Test
+    public void shouldRegisterPythonUdfFromPyFile() throws IOException {
+        String pathFile = getPath("test_udf.py");
+        String sqlRegisterUdf = "CREATE TEMPORARY FUNCTION TEST_UDF AS 'test_udf.test_udf' LANGUAGE PYTHON";
+
+        when(tableEnvironment.getConfig()).thenReturn(tableConfig);
+        when(tableConfig.getConfiguration()).thenReturn(configuration);
+        when(pythonUdfConfig.getPythonFiles()).thenReturn(pathFile);
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", pathFile);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterUdf);
+    }
+
+    @Test
+    public void shouldRegisterPythonUdfFromPyAndZipFile() throws IOException {
+        String zipPathFile = getPath("python_udf.zip");
+        String pyPathFile = getPath("test_udf.py");
+
+        String sqlRegisterFirstUdf = "CREATE TEMPORARY FUNCTION ADD AS 'python_udf.scalar.add.add' LANGUAGE PYTHON";
+        String sqlRegisterSecondUdf = "CREATE TEMPORARY FUNCTION SUBSTRACT AS 'python_udf.vectorized.substract.substract' LANGUAGE PYTHON";
+        String sqlRegisterThirdUdf = "CREATE TEMPORARY FUNCTION TEST_UDF AS 'test_udf.test_udf' LANGUAGE PYTHON";
+
+        when(tableEnvironment.getConfig()).thenReturn(tableConfig);
+        when(tableConfig.getConfiguration()).thenReturn(configuration);
+        when(pythonUdfConfig.getPythonFiles()).thenReturn(zipPathFile + "," + pyPathFile);
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", zipPathFile + "," + pyPathFile);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterFirstUdf);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterSecondUdf);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterThirdUdf);
+    }
+
+    @Test
+    public void shouldThrowIOExceptionIfPythonFilesNotInZipOrPyFormat() throws IOException {
+        expectedEx.expect(IOException.class);
+        expectedEx.expectMessage("Python files should be in .py or .zip format");
+
+        File file = File.createTempFile("test_file", ".txt");
+        file.deleteOnExit();
+
+        when(tableEnvironment.getConfig()).thenReturn(tableConfig);
+        when(tableConfig.getConfiguration()).thenReturn(configuration);
+        when(pythonUdfConfig.getPythonFiles()).thenReturn("test_file.txt");
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionIfPythonFilesNotExist() throws IOException {
```

Comment:
can we handle this with proper exception? instead of throwing null pointer?

Reply:
whats should be the proper exception? if PYTHON_UDF_ENABLE is true then PYTHON_FILES are required. it will throw  NullPointerException.

Reply:
should throw something like, PythonFilesNotFoundException

Reply:
ok, will create new exception PythonFilesNotFoundException to handle that